solr/core/src/java/org/apache/solr/search/grouping/endresulttransformer/GroupedEndResultTransformer.java

```diff
@@ -75,7 +75,13 @@ public void transform(Map<String, ?> result, ResponseBuilder rb, SolrDocumentSou
           SimpleOrderedMap<Object> groupResult = new SimpleOrderedMap<>();
           if (group.groupValue != null) {
             // use createFields so that fields having doc values are also supported
-            List<IndexableField> fields = groupField.createFields(group.groupValue.utf8ToString());
+            final String groupValue;
+            if (rb.getGroupingSpec().isSkipSecondGroupingStep()) {
+              groupValue = groupField.getType().indexedToReadable(group.groupValue.utf8ToString());
```

Comment:
I struggled (again) to comprehend why this change is needed. https://github.com/cpoerschke/lucene-solr/commit/20129e7d3f7e12f442254e780e7da9a590a9036b proposes to factor out a local `bytesRefToString` functor with fairly detailed comments. What do you think?

Reply:
Looks good, merged